cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+    }
+
+    private void logCurrentSourcePartitions(List<String> sourcePartitionList, String processId, String instanceExecId, String table) {
+        ProcessLog processLog = new ProcessLog();
+        List<ProcessLogInfo> partitionLogInfoList = new ArrayList<>();
+        for (String sourcePartition : sourcePartitionList) {
+            ProcessLogInfo processLogInfo = new ProcessLogInfo();
+            processLogInfo.setAddTs(new Date());
+            processLogInfo.setProcessId(Integer.parseInt(processId));
+            processLogInfo.setMessage(sourcePartition);
+            processLogInfo.setInstanceRef(Long.parseLong(instanceExecId));
+            processLogInfo.setLogCategory("C2C");
+            processLogInfo.setMessageId(table + " partition");
+            partitionLogInfoList.add(processLogInfo);
+        }
+        processLog.logList(partitionLogInfoList);
+    }
+
+    private void logCurrentSourceColumns(List<String> sourceColumnList, String processId, String instanceExecId, String table) {
+        ProcessLog processLog = new ProcessLog();
+        List<ProcessLogInfo> columnLogInfoList = new ArrayList<>();
+        for (String sourceColumn : sourceColumnList) {
+            ProcessLogInfo processLogInfo = new ProcessLogInfo();
+            processLogInfo.setAddTs(new Date());
+            processLogInfo.setProcessId(Integer.parseInt(processId));
+            processLogInfo.setMessage(sourceColumn);
+            processLogInfo.setInstanceRef(Long.parseLong(instanceExecId));
+            processLogInfo.setLogCategory("C2C");
+            processLogInfo.setMessageId(table + " column");
+            columnLogInfoList.add(processLogInfo);
+        }
+        processLog.logList(columnLogInfoList);
+    }
+
+    private void formStageAndDestTableDDLs(Statement st, List<String> sourceColumnList, String sourceDb, String destDb, String table, String sourceStgtable, String bdreTechPartition, String processId,String instanceExecId) throws Exception {
+        ResultSet rsPartitionList = st.executeQuery("desc "+table);
+        int index = 0;
+        StringBuffer partitionList = new StringBuffer("");
+        List<String> sourcePartitionColumnList = new ArrayList<>();
+        while (rsPartitionList.next()) {
+            if (rsPartitionList.getString(1).equals("# Partition Information")) {
+                index++;
+            }
+            if (index > 0) index++;
+            if (index > 4) {
+                String partitionNameAndDataType = rsPartitionList.getString(1) + " " + rsPartitionList.getString(2).toUpperCase();
+                partitionList.append(partitionNameAndDataType + ",");
+                sourcePartitionColumnList.add(partitionNameAndDataType);
+            }
+        }
+
+        sourceColumnList.removeAll(sourcePartitionColumnList);
+        //removing partition columns from the total columns list and concatenating with commas
+        StringBuffer finalColumns = new StringBuffer("");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1149)